src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -52,17 +59,27 @@ public static VariantAnnotatorEngine ofAllMinusExcluded(final List<String> annot
      * @param annotationsToExclude list of annotations to exclude
      * @param dbSNPInput input for variants from a known set from DbSNP or null if not provided.
      *                   The annotation engine will mark variants overlapping anything in this set using {@link htsjdk.variant.vcf.VCFConstants#DBSNP_KEY}.
+     * @param featureInputs list of inputs with known variants.
+     *                   The annotation engine will mark variants overlapping anything in those sets using the name given by {@link FeatureInput#getName()}.
      */
-    public static VariantAnnotatorEngine ofSelectedMinusExcluded(final List<String> annotationGroupsToUse, final List<String> annotationsToUse, final List<String> annotationsToExclude, final FeatureInput<VariantContext> dbSNPInput) {
+    public static VariantAnnotatorEngine ofSelectedMinusExcluded(final List<String> annotationGroupsToUse,
+                                                                 final List<String> annotationsToUse,
+                                                                 final List<String> annotationsToExclude,
+                                                                 final FeatureInput<VariantContext> dbSNPInput,
+                                                                 final List<FeatureInput<VariantContext>> featureInputs) {
         Utils.nonNull(annotationGroupsToUse, "annotationGroupsToUse is null");
         Utils.nonNull(annotationsToUse, "annotationsToUse is null");
         Utils.nonNull(annotationsToExclude, "annotationsToExclude is null");
-        return new VariantAnnotatorEngine(AnnotationManager.ofSelectedMinusExcluded(annotationGroupsToUse, annotationsToUse, annotationsToExclude), dbSNPInput);
+        Utils.nonNull(featureInputs, "featureInputs is null");
+        return new VariantAnnotatorEngine(AnnotationManager.ofSelectedMinusExcluded(annotationGroupsToUse, annotationsToUse, annotationsToExclude), dbSNPInput, featureInputs);
     }
 
-    private VariantOverlapAnnotator initializeOverlapAnnotator(final FeatureInput<VariantContext> dbSNPInput) {
+    private VariantOverlapAnnotator initializeOverlapAnnotator(final FeatureInput<VariantContext> dbSNPInput, final List<FeatureInput<VariantContext>> featureInputs) {
         final Map<FeatureInput<VariantContext>, String> overlaps = new LinkedHashMap<>();
-        if ( dbSNPInput != null ) {
+        for ( final FeatureInput<VariantContext> fi : featureInputs) {
+            overlaps.put(fi, fi.getName());
+        }
+        if ( dbSNPInput != null && ! overlaps.values().contains(VCFConstants.DBSNP_KEY)) {
```

Comment:
What does it mean if `overlaps.values().contains(VCFConstants.DBSNP_KEY)` is true at this point? Doesn't it mean that the caller put the DBSNP `FeatureInput` into the `featureInputs` List, and also passed a DBSNP input in separately? What if these two DBSNP inputs were not the same? Then we'd silently ignore the one that was passed in individually, which doesn't seem like very good behavior.

Recommend that you enforce a requirement that the DBSNP input be passed in separately and not via the List (ie., throw here if `overlaps.values().contains(VCFConstants.DBSNP_KEY)` is true).